docs/v1-proposed-api.md

```diff
@@ -0,0 +1,145 @@
+# Proposed Changes for Hive V1 API
+
+## ClusterDeployment
+
+ 1. Require an InstallConfig ConfigMap, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
```

Comment:
Since the InstallConfig contains secret information, this should probably be a `Secret` instead of a `ConfigMap`.